onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,592 @@
+                return false;
+
+        }
+    }
+
+    if(node->OpType() == "Softmax"){
+
+        //First dimension of Softmax input has to be 1
+        if(input_dims != 0 ){
+            if(node_inputs[0]->Shape()->dim(0).dim_value() != 1)
+                return false;
+        }
+
+        //Only 2D input supported on MYRIAD and HDDL
+        if(dev_id == "MYRIAD" || dev_id == "HDDL"){
+            if(input_dims != 2)
+                return false;
+        }
+    }
+
+    //Only 2D MatMul is supported
+    if(node->OpType() == "MatMul"){
+        for(size_t i = 0; i < node_inputs.size(); i++){
+
+            if(node_inputs[i]->Shape() != nullptr){
+                if(node_inputs[i]->Shape()->dim_size() != 2)
+                    return false;
+            }
+        }
+    }
+
+    if(node->OpType() == "Flatten"){
+
+        if(dev_id == "MYRIAD" || dev_id == "HDDL"){
+            if(node_inputs[0]->Shape() != nullptr && node_inputs[0]->Shape()->dim(0).dim_value() != 1)
+                return false;
+        }
+    }
+
+    return true;
+}
+
+//Checks whether the node is supported by OpenVINO
+bool IsOpSupported(std::string name){
+
+    std::set<std::string> supported_ops = {
+        "Add",
+        "BatchNormalization",
+        "Conv",
+        "GlobalAveragePool",
+        "Relu",
+        "Reshape",
+        "Flatten",
+        "Gemm",
+        "MaxPool",
+        "AveragePool",
+        "Concat",
+        "Dropout",
+        "LRN",
+        "Softmax",
+        "Mul",
+        "Sum",
+        "Transpose",
+        "Identity",
+        "MatMul",
+        "Unsqueeze",
+        "ImageScaler",
+        "LeakyRelu",
+        "GlobalMaxPool"};
+
+    auto iter = supported_ops.find(name);
+    return iter != supported_ops.end();
+}
+
+
+//Checks if the entire graph is supported by OpenVINO EP and returns false if it is not.
+
+bool IsGraphSupported(const onnxruntime::GraphViewer& graph_viewer, std::string dev_id){
+
+  const auto& initializers = graph_viewer.GetAllInitializedTensors();
+
+  auto node_indexes = graph_viewer.GetNodesInTopologicalOrder();
+
+  auto model_proto = GetModelProtoFromFusedNode(graph_viewer);
+
+  auto graph_proto = model_proto.mutable_graph();
+  int input_dims = 0;
+  int output_dims = 0;
+  int num_inputs = graph_viewer.GetInputs().size();
+  int num_outputs = graph_viewer.GetOutputs().size();
+
+  if (num_inputs != 0)
+    input_dims = graph_proto->input(0).type().tensor_type().shape().dim_size();
+
+  if (num_outputs != 0)
+    output_dims = graph_proto->output(0).type().tensor_type().shape().dim_size();
+
+  //GPU Plugin does not support single dimensional input and 5 dimensional input
+  if (dev_id == "GPU") {
+    if (input_dims == 1 || input_dims == 5 || output_dims == 5)
+      return false;
+  }
+
+  for (auto index : node_indexes) {
+    const auto node = graph_viewer.GetNode(index);
+
+    if(node == nullptr)
+        return false;
```

Comment:
continue, not return false.

Reply:
False is returned here, as we have assumed there to be an error in the graph since it is returning a nullptr. Such errors might cause issues with Model optimizer or at the Compile().
Is this assumption that a nullptr indicates graph error valid? would it be safe to ignore the node with nullptr and continue ahead? Could you describe a case where a nullptr may be returned by GetNode() ?

Reply:
looked into this more.
technically, GetNode() api can return a nullptr (when graph is edited, say from GraphTransforms) but in this case, we shouldn't have any nullptr because we are getting the node indexes from GetNodesInTopologicalOrder() which should only return valid nodes. sorry for the confusion.

Reply:
you can remove this node == nullptr check.

Reply:
cool. We will update the PR with that commit